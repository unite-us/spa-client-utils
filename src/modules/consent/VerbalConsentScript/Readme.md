```jsx
  <VerbalConsentScript
    title="Verbal Informed Consent for Participation and Information Sharing"
    showBaseCard
  />
```
