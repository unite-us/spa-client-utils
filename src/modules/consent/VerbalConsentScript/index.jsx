import React from 'react';
import PropTypes from 'prop-types';
import { BaseCard, BaseCardBody, BaseCardHeader } from '@unite-us/ui';

const CONSENT_EMAIL_URL = 'mailto:consent@uniteus.com';
const PRIVACY_URL = 'https://www.uniteus.com/privacy';

const ConsentTextScript = ({ openLink, title }) => (
  <div className="consent-text-script">
    <h3 className="consent-text-script__title mb-one">{title}</h3>
    <p className="mb-one">
      By consenting, you agree to share information with a Network of health and social service partners powered by
      Unite Us software. This Network is made up of entities and individuals who are directly involved in your care or payment of care.
      Your personal information may be shared securely on the Network in accordance with privacy laws
      to connect you with services.
    </p>

    <p className="mb-one">
      This consent covers all information shared by you or by anyone that has the right to share information on your
      behalf and is relevant to the recipient&#39;s involvement in your care or payment for your care. You can always limit the information you provide on the Network by requesting to have it removed.
    </p>

    <p className="mb-one">
      To understand how your information may be used and kept safe on the Network, please see
      <a
        className="ml-quarter"
        onClick={() => openLink(PRIVACY_URL)}
        rel="noopener noreferrer"
        role="link"
        tabIndex={0}
        target="_blank"
      >uniteus.com/privacy</a>.
    </p>

    <p className="mb-one">
      If you no longer want your information shared on the Network, you can email
      <a
        className="ml-quarter mr-quarter"
        onClick={() => openLink(CONSENT_EMAIL_URL)}
        rel="noopener noreferrer"
        role="link"
        tabIndex={0}
        target="_blank"
      >
        consent@uniteus.com
      </a>
      or ask any Network partner.
    </p>

    <p className="verbal-consent-script__heavy mb-one">
      Please answer the following questions with a Yes or a No:
    </p>

    <ul>
      <li>
        Do you provide your consent? [Client or Personal Representative / Guardian (if applicable) must say: Yes]
      </li>
      <li>
        Do you understand and agree with what I have just read? [Yes]
      </li>
      <li>
        Did I give you a chance to ask any questions that you had? [Yes]
      </li>
      <li>
        Do you have any other questions? [No]
      </li>
    </ul>
  </div>
);

ConsentTextScript.propTypes = {
  openLink: PropTypes.func.isRequired,
  title: PropTypes.string,
};

ConsentTextScript.defaultProps = {
  title: '',
};

const VerbalConsentScript = ({ openLink, showBaseCard, title }) => {
  if (showBaseCard) {
    return (
      <div className="verbal-consent-script container">
        <BaseCard>
          <BaseCardHeader
            className="verbal-consent-script__title"
            title={title}
          />
          <BaseCardBody
            className="verbal-consent-script__body"
            withPadding
          >
            <ConsentTextScript openLink={openLink} />
          </BaseCardBody>
        </BaseCard>
      </div>
    );
  }

  return (
    <div className="verbal-consent-script container">
      <ConsentTextScript openLink={openLink} title={title} />
    </div>
  );
};

VerbalConsentScript.propTypes = {
  openLink: PropTypes.func,
  showBaseCard: PropTypes.bool,
  title: PropTypes.string,
};

VerbalConsentScript.defaultProps = {
  openLink: url => window.open(url, '_blank'),
  showBaseCard: false,
  title: 'Script for Consent to Participate in the Unite Us Network',
};

export default VerbalConsentScript;
