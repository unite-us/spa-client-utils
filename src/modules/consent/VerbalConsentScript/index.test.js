import React from 'react';
import renderer from 'react-test-renderer';
import VerbalConsentScript from './';

describe('VerbalConsentScript', () => {
  it('renders', () => {
    const props = {
      showBaseCard: false,
      title: 'Consent Form',
    };

    const comp = renderer.create(<VerbalConsentScript {...props} />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it('renders Basecard', () => {
    const props = {
      showBaseCard: true,
      title: 'Consent Form',
    };

    const comp = renderer.create(<VerbalConsentScript {...props} />).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
