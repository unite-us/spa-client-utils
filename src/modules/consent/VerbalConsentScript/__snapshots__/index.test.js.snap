// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerbalConsentScript renders 1`] = `
<div
  className="verbal-consent-script container"
>
  <div
    className="consent-text-script"
  >
    <h3
      className="consent-text-script__title mb-one"
    >
      Consent Form
    </h3>
    <p
      className="mb-one"
    >
      By consenting, you agree to share information with a Network of health and social service partners powered by Unite Us software. This Network is made up of entities and individuals who are directly involved in your care or payment of care. Your personal information may be shared securely on the Network in accordance with privacy laws to connect you with services.
    </p>
    <p
      className="mb-one"
    >
      This consent covers all information shared by you or by anyone that has the right to share information on your behalf and is relevant to the recipient's involvement in your care or payment for your care. You can always limit the information you provide on the Network by requesting to have it removed.
    </p>
    <p
      className="mb-one"
    >
      To understand how your information may be used and kept safe on the Network, please see
      <a
        className="ml-quarter"
        onClick={[Function]}
        rel="noopener noreferrer"
        role="link"
        tabIndex={0}
        target="_blank"
      >
        uniteus.com/privacy
      </a>
      .
    </p>
    <p
      className="mb-one"
    >
      If you no longer want your information shared on the Network, you can email
      <a
        className="ml-quarter mr-quarter"
        onClick={[Function]}
        rel="noopener noreferrer"
        role="link"
        tabIndex={0}
        target="_blank"
      >
        consent@uniteus.com
      </a>
      or ask any Network partner.
    </p>
    <p
      className="verbal-consent-script__heavy mb-one"
    >
      Please answer the following questions with a Yes or a No:
    </p>
    <ul>
      <li>
        Do you provide your consent? [Client or Personal Representative / Guardian (if applicable) must say: Yes]
      </li>
      <li>
        Do you understand and agree with what I have just read? [Yes]
      </li>
      <li>
        Did I give you a chance to ask any questions that you had? [Yes]
      </li>
      <li>
        Do you have any other questions? [No]
      </li>
    </ul>
  </div>
</div>
`;

exports[`VerbalConsentScript renders Basecard 1`] = `
<div
  className="verbal-consent-script container"
>
  <div
    className="ui-base-card ui-base-card--bordered"
  >
    <div
      className="ui-base-card-header with-border verbal-consent-script__title"
    >
      <div
        className="ui-base-card-header__main-content"
      >
        <h2
          className="ui-base-card-header__title"
        >
          Consent Form
        </h2>
        <div
          className="ui-base-card-header__children"
        />
      </div>
    </div>
    <div
      className="ui-gradient"
    >
      <div
        className="ui-base-card-body ui-base-card-body--with-padding verbal-consent-script__body"
      >
        <div
          className="consent-text-script"
        >
          <h3
            className="consent-text-script__title mb-one"
          >
            
          </h3>
          <p
            className="mb-one"
          >
            By consenting, you agree to share information with a Network of health and social service partners powered by Unite Us software. This Network is made up of entities and individuals who are directly involved in your care or payment of care. Your personal information may be shared securely on the Network in accordance with privacy laws to connect you with services.
          </p>
          <p
            className="mb-one"
          >
            This consent covers all information shared by you or by anyone that has the right to share information on your behalf and is relevant to the recipient's involvement in your care or payment for your care. You can always limit the information you provide on the Network by requesting to have it removed.
          </p>
          <p
            className="mb-one"
          >
            To understand how your information may be used and kept safe on the Network, please see
            <a
              className="ml-quarter"
              onClick={[Function]}
              rel="noopener noreferrer"
              role="link"
              tabIndex={0}
              target="_blank"
            >
              uniteus.com/privacy
            </a>
            .
          </p>
          <p
            className="mb-one"
          >
            If you no longer want your information shared on the Network, you can email
            <a
              className="ml-quarter mr-quarter"
              onClick={[Function]}
              rel="noopener noreferrer"
              role="link"
              tabIndex={0}
              target="_blank"
            >
              consent@uniteus.com
            </a>
            or ask any Network partner.
          </p>
          <p
            className="verbal-consent-script__heavy mb-one"
          >
            Please answer the following questions with a Yes or a No:
          </p>
          <ul>
            <li>
              Do you provide your consent? [Client or Personal Representative / Guardian (if applicable) must say: Yes]
            </li>
            <li>
              Do you understand and agree with what I have just read? [Yes]
            </li>
            <li>
              Did I give you a chance to ask any questions that you had? [Yes]
            </li>
            <li>
              Do you have any other questions? [No]
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
